index_configuration_tool/Dockerfile

```diff
@@ -1,20 +1,17 @@
-# Multistage build - 1st stage
-FROM python:3.11 AS builder
+FROM opensearch-data-prepper:2.4.0-SNAPSHOT
 COPY requirements.txt .
 
 # Install dependencies to local user directory
+RUN apk update
+RUN apk add --no-cache python3 py-pip
 RUN pip install --user -r requirements.txt
 
-# Multistage build - 2nd stage
-FROM python:3.11-slim
 WORKDIR /code
-# Copy only required dependencies
-COPY --from=builder /root/.local /root/.local
 # Copy only source code
 COPY ./*.py .
 
 # update PATH
 ENV PATH=/root/.local:$PATH
 
 # make sure you include the -u flag to have our stdout logged
-ENTRYPOINT [ "python", "-u", "./main.py" ]
+CMD python -u ./main.py -r $DATA_PREPPER_PATH/pipelines/pipelines.yaml $DATA_PREPPER_PATH/pipelines/pipelines.yaml; $DATA_PREPPER_PATH/bin/data-prepper
```

Comment:
This is more of a me-and-docker question, but why switch from ENTRYPOINT to CMD?

Reply:
> This is more of a me-and-docker question, but why switch from ENTRYPOINT to CMD?

Good question! I was using `CMD` because the [base Data Prepper Dockerfile](https://github.com/kartg/data-prepper/blob/main/release/docker/Dockerfile) uses `CMD`, but it turns out [ENTRYPOINT in the child image overrides the parent CMD](https://stackoverflow.com/a/49031590). Updated.

> This seems to be largely because we're reading from and writing to the same YAML file, and then running with that YAML file. Would the situation be "safer" if we e.g. renamed pipelines.yaml to original-pipelines.yaml, used that as the source for ./main.py but output to pipelines.yaml and then ran data-prepper? Would that ensure that only the correct indices get migrated?

Agree that this is a reasonable mitigation. I've changed the path for the input YAML file for ICT to not live in the data-prepper `pipelines` directory. While this mitigates this bug, the Docker image still runs the commands in sequence, which causes Data Prepper to start but fail because no pipeline configuration is found. I'm tracking a fix for this in https://opensearch.atlassian.net/browse/MIGRATIONS-1218